### PR TITLE
Include all Teleport CAs in the rotation guide

### LINKED ISSUE #37125: Document CA rotation steps
## Applies To

https://goteleport.com/docs/management/operations/ca-rotation/

## Details

The documentation doesn't mention what Ca rotation implies for most of Teleport protocols.

To properly rotate a CA, you must list and update every reference to the previous CA. SSH access and agents are the happy path as it's happening automatically. Other protocols are way more complex and this is not reflected in the docs.

Rotating the CA will break Database Access, AWS console access, agentless OpenSSH nodes, potentially App Access.

The user must list every resource relying on the previous Teleport CA and have them trust the new CA before finishing the rotation.

## How will we know this is resolved?

A user using Teleport for all protocols (SSH, SSH agentless, Windows, Database, App, AWS console, Kubernetes, ...) can follow the CA rotation documentation without breaking their Teleport cluster and loosing access to their resources.

## Related Issues

I think this issue blocks the Helm chart cloud KMS support by default for GCP and AWS. When we'll add support in the chart, 
users will switch to KMS, which implies going a CA rotation. If the documentation is not accurate, they will end up losing access to their Teleport resources.

### LINKED ISSUE #44398: Procedure for OpenSSH Cert Authority Rotation
## Applies To
https://goteleport.com/docs/management/operations/ca-rotation/

## Details
Document clear and detailed instructions for handling  CA rotation for OpenSSH nodes. Docs should also clarify the phase during the rotation cycle when the usage of the one-shot command `teleport join openssh` should be executed.

something similar to the DB cert  rotation docs
https://goteleport.com/docs/management/operations/db-ca-rotation/

## How will we know this is resolved?
Guide should detail steps such that after a CA rotation, access to OpenSSH nodes via tsh is successful.

### PR DESCRIPTION
**Pull request by ptgott**
Closes #37125
Closes #44398

This change clarifies the differences between Teleport CAs in the CA Rotation guide. Since the Database CA Rotation guide is almost identical to the original CA Rotation guide, this change merges the two guides.

To include relevant information for each CA, this change adds a step to help the user choose a CA to rotate, and includes specific information for each CA in that step.

Another possibility would be to spread information for each CA across the instructions for each transition phase. However, since each CA requires a different set of checks/updates in different phases, it makes sense to make the instructions for each CA more free-flowing while restricting them to a single, early section.

Most information about checking the status of a rotation in the original guide had to do with the `host` CA, and this change moves that information into an H3 related to that CA.

### CODE REVIEW

**Comment by ptgott** [comment]
Hi @GavinFrazar, it'd be great to get a review from you if you have time, since you wrote the original database CA rotation guide and have done recent work on our CAs. Thanks!

**Comment by GavinFrazar** [comment]
> Hi @GavinFrazar, it'd be great to get a review from you if you have time, since you wrote the original database CA rotation guide and have done recent work on our CAs. Thanks!

Awesome! I was about to propose essentially the same thing as this, but you beat me by 3 weeks lol

I'll try to take a look today as I'll be on vacation all next week

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:9]: I would say that Teleport uses the certificate authorities to _issue certificates_. Verification against the CA is done by the target resources (Windows desktop, database, SSH node, etc)

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:63]:
```suggestion
|CA type|Certificates it issues|
```

Perhaps?

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:77]:
```suggestion
The `host` CA issues certificates for Teleport cluster components (agents
 and Proxy Service
instances and the Teleport Auth Service).
```

**Review by zmb3 — COMMENTED** [review]
My comments apply throughout the whole doc and may be a matter of preference, but I would frame the doc around what certificates each CA issues instead of around what traffic it authenticates.

(Technically, CAs don't really authenticate anything. They issue certs. The entities communicating with those certs are responsible for authenticating each other.)

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:65]: table rendering fix:
```suggestion
|---|---|---|
```

Comment (GavinFrazar) [reply docs/pages/management/operations/ca-rotation.mdx:65]: I think we should drop the "verified by" column.

Accurately describing and enumerating the things that verify certs issued by a given CA is too much info.

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:31]:
```suggestion
1. `update_servers`: Teleport cluster components (Agents, Auth Service, and Proxy Service
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:355]:
```suggestion
Initiate the `update_servers` phase. In this phase, Teleport cluster components (Agents,
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:442]:
```suggestion
Complete a rotation in manual mode first to understand all the edge-cases and
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:440]: @zmb3 what do you think about a code change to just reject automatic rotation requests for several of our CAs? For example, I can't think of a legit use case for doing automatic rotation with `db` or `db_client` CAs, because you have to reissue all of your database certificates or reconfigure them to trust the new CA cert. An automatic rotation will just break access to self-hosted databases like a ticking time bomb.

As far as I know, host and user CA automatic rotation is fine (maybe not user if there are desktops configured?). Not sure about other ones.

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:149]:
```suggestion
trust certificates issued by the CA. Self-hosted databases can be issued a certificate by the `db` CA, which the Database Service uses by default to verify that a database
server is a genuine Teleport-protected resource.
```

it's optional to use a `db` CA cert - they can their own custom CA to issue a cert for their databases, but they must configure the db service to trust their custom CA

**Review by GavinFrazar — APPROVED** [review]
I have a few suggestions but nothing worth blocking the PR for.

By the way you need to remove all links to the deleted db-ca-rotation.mdx file (one in operations.mdx, the other in db-ca-migrations.mdx)

**Comment by ptgott** [comment]
@zmb3 Putting this on your radar for when you have some time to review. Thanks!

**Comment by ptgott** [comment]
@zmb3 Following up on this one for when you have time for another look. Thanks!

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:38]:
```suggestion
Before the final `standby` phase, you can also put the rotation in the `rollback`
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:62]:
```suggestion
exception is rotating the `db` and `db_client` CAs. These two CAs must
be rotated together.
```

Not sure if my language here is exactly right, but we should explain what the exception we refer to is.

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:67]: The _subjects_ in the certificates issued by these CAs are still Teleport users, not Teleport services (those still use the host CA).

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:69]: Is this correct? The Teleport Database Service is a server, right? Seems odd that a server cert would be issued by a client CA?

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:71]: Perhaps?

```suggestion
|[`jwt`](#jwt)|Teleport users accessing web applications|
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:109]:
```suggestion
| Kubernetes Service  | `kube_server`    |
| Windows Desktop Service | `windows_desktop_service` |
```

Assuming rotation status is included for w_d_s

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:119]:
```suggestion
If you are joining Teleport processes to a cluster by dialing the Teleport Auth Service directly,
```

Not sure what the join token has to do with this. All joining requires a join token.

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:128]:
```suggestion
also signs client certificates for users connecting to Windows desktops and Teleport SSH servers.
Teleport-protected servers and Windows desktops use these certificates
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:136]:
```suggestion
guide](../../desktop-access/active-directory-manual.mdx) to export the
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:243]:
```suggestion
The `oidc_idp` CA signs messages sent by the Teleport OIDC AWS integration. AWS
```

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:245]: I think we removed EICE support from Teleport, didn't we?

File: docs/pages/management/operations/ca-rotation.mdx
Comment (zmb3) [inline docs/pages/management/operations/ca-rotation.mdx:250]: We are in the process of removing the S3 bucket, so I think we can remove this.

**Review by zmb3 — COMMENTED** [review]
Sorry for losing track of this one!

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [reply docs/pages/management/operations/ca-rotation.mdx:69]: This is correct, the Teleport Database Service is issued a client cert by the `db_client` CA at connection time, which it then uses to authenticate itself to the database as a client.

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [reply docs/pages/management/operations/ca-rotation.mdx:245]: yes, all the code is still there but we removed the tile from the web ui

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [reply docs/pages/management/operations/ca-rotation.mdx:243]: We will expand the OIDC integration to other clouds, I think Azure is already in the works.

I would stick with the generic IdP language here, maybe even scrap the AWS specific language and speak generically about "the cloud" or "the relying party"?

File: docs/pages/management/operations/ca-rotation.mdx
Comment (GavinFrazar) [inline docs/pages/management/operations/ca-rotation.mdx:249]: nit: I would mention `/.well-known/openid-configuration` as well, which includes the path to the jwks uri, e.g. mine looks like:
```json
{
  "issuer": "https://alpha.devteleport.com",
  "jwks_uri": "https://alpha.devteleport.com/.well-known/jwks-oidc",
  "claims": [
    "iss",
    "sub",
    "obo",
    "aud",
    "jti",
    "iat",
    "exp",
    "nbf"
  ],
  "id_token_signing_alg_values_supported": [
    "RS256"
  ],
  "response_types_supported": [
    "id_token"
  ],
  "scopes_supported": [
    "openid"
  ],
  "subject_types_supported": [
    "public",
    "pair-wise"
  ]
}
```

**Comment by ptgott** [comment]
@zmb3 Following up to see if my latest changes satisfy your feedback. Thanks!

**Comment by ptgott** [comment]
Hi @zmb3, checking again to see if I have resolved your feedback. Thanks!

**Review by zmb3 — APPROVED** [review]

**Pull request merged**
